### PR TITLE
add cybergear macro25 keyboard

### PR DESCRIPTION
**Pull request by CyberGear**
Adding new  keyboard: cybergear macro25 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cybergear/macro25/rules.mk [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+# generated by KBFirmware JSON to QMK Parser
+# https://noroadsleft.github.io/kbf_qmk_converter/
+
+VIA_ENABLE = yes
```
Comment (Duckle29) [inline keyboards/cybergear/macro25/rules.mk:27]:
```suggestion
```

Via support should be done through a VIA keymap such as here: https://github.com/qmk/qmk_firmware/tree/master/keyboards/keebio/bdn9/keymaps/via

**Review by Duckle29 — CHANGES_REQUESTED** [review]
A few things need changed. You got a unique VID PID so that's dope, but you are missing a default keymap (named default), and you also don't have an info.json as it says in the PR checklist.

As mentioned in a separate comment, via support has to be implemented as a separate keymap.

File: keyboards/cybergear/macro25/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// https://codepen.io/mvaneijgen/full/LYEVQXp
+
+#define MAC(kc) (QK_RCTL | QK_RGUI | QK_RALT | (kc))
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        MAC(KC_KP_0), MAC(KC_KP_1), MAC(KC_KP_2), MAC(KC_KP_3), MAC(KC_KP_4),
+        MAC(KC_KP_5), MAC(KC_KP_6), MAC(KC_KP_7), MAC(KC_KP_8), MAC(KC_KP_9)
+    )
+};
```
Comment (Duckle29) [inline keyboards/cybergear/macro25/keymaps/via/keymap.c:28]: The VIA keymap should have as many layers as used by dynamic keymaps. Default is 4
https://caniusevia.com/docs/configuring_qmk#create-a-keymapc-in-keyboardskeyboardnamekeymapsvia

File: keyboards/cybergear/macro25/keymaps/via/via-macro25.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    "rows": 2,
+    "cols": 5
+  },
+  "layouts": {
+    "keymap": [
+      [ "0,0", "0,1", "0,2", "0,3", "0,4" ],
+      [ "1,0", "1,1", "1,2", "1,3", "1,4" ]
+    ]
+  }
+}
```
Comment (Duckle29) [inline keyboards/cybergear/macro25/keymaps/via/via-macro25.json:16]: This json is for the VIA side over here https://github.com/the-via/keyboards/ It shouldn't be in QMK :)

**Review by Duckle29 — CHANGES_REQUESTED** [review]
Aside from the specific notes. You still need an info.json. It's different from the via json (which as noted is for via ;) )

https://docs.qmk.fm/#/reference_info_json 

You can test it by importing it on https://config.qmk.fm by hitting ctrl+shift+i

**Review by Duckle29 — APPROVED** [review]
LGTM

File: keyboards/cybergear/macro25/keymaps/via/rules.mk [hunk 4]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (fauxpark) [inline keyboards/cybergear/macro25/keymaps/via/rules.mk:1]:
```suggestion
```

File: keyboards/cybergear/macro25/macro25.c [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "macro25.h"
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
+
+
```
Comment (fauxpark) [inline keyboards/cybergear/macro25/macro25.c:22]:
```suggestion
```

File: keyboards/cybergear/macro25/macro25.h [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    K00, K01, K02, K03, K04, \
+    K05, K06, K07, K08, K09  \
+) { \
+    { K00, K01, K02, K03, K04 }, \
+    { K05, K06, K07, K08, K09 }, \
+}
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
+
```
Comment (fauxpark) [inline keyboards/cybergear/macro25/macro25.h:31]:
```suggestion
```

File: keyboards/cybergear/macro25/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+Macro keyboard meade using arduino pro micro, with VIA support
+
+* Keyboard Maintainer: [CyberGeear](https://github.com/CyberGear)
+* Hardware Supported: arduino pro micro
+* Hardware Availability: [repo](https://github.com/CyberGear/mk-macro-01)
+
+compiling (after setting up your qmk build environment):
+
+    qmk compile -kb cybergear/macro25 -km CyberGear
```
Comment (fauxpark) [inline keyboards/cybergear/macro25/readme.md:13]:
```suggestion
Make example for this keyboard (after setting up your build environment):

    make cybergear/macro25:default
```

File: keyboards/cybergear/macro25/readme.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+# CyberGeear Macro25
+
+![image](https://github.com/CyberGear/mk-macro-01/raw/master/images/result-preview.jpg)
```
Comment (fauxpark) [inline keyboards/cybergear/macro25/readme.md:3]:
```suggestion
![](https://i.imgur.com/Jk2xGMvh.jpg)
```

File: keyboards/cybergear/macro25/readme.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# CyberGeear Macro25
```
Comment (fauxpark) [inline keyboards/cybergear/macro25/readme.md:1]:
```suggestion
# CyberGear Macro25
```

File: keyboards/cybergear/macro25/readme.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+* Keyboard Maintainer: [CyberGeear](https://github.com/CyberGear)
+* Hardware Supported: arduino pro micro
+* Hardware Availability: [repo](https://github.com/CyberGear/mk-macro-01)
+
+compiling (after setting up your qmk build environment):
+
+    qmk compile -kb cybergear/macro25 -km CyberGear
+
+See the [Setting Up Your QMK Environment](https://docs.qmk.fm/#/newbs_getting_started)
```
Comment (fauxpark) [inline keyboards/cybergear/macro25/readme.md:15]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cybergear/macro25/config.h [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+#define PRODUCT         Macro25 (mk-macro-01)
+
+#define MATRIX_ROWS 2
+#define MATRIX_COLS 5
+
+#define MATRIX_ROW_PINS { E6, B4 }
+#define MATRIX_COL_PINS { B1, F7, F6, F5, F4 }
+
+#define DIODE_DIRECTION COL2ROW
+
```
Comment (fauxpark) [inline keyboards/cybergear/macro25/config.h:35]: Fairly certain everything except `MANUFACTURER` and `PRODUCT` can be removed here, as it is already in info.json.

Comment (CyberGear) [reply keyboards/cybergear/macro25/config.h]: Removed all mentioned in 
```
Making cybergear/macro25 with keymap default

⚠ cybergear/macro25: DEVICE_VER in config.h is overwriting usb.device_ver in info.json
⚠ cybergear/macro25: DIODE_DIRECTION in config.h is overwriting diode_direction in info.json
⚠ cybergear/macro25: PRODUCT_ID in config.h is overwriting usb.pid in info.json
⚠ cybergear/macro25: VENDOR_ID in config.h is overwriting usb.vid in info.json
⚠ cybergear/macro25: Matrix pins are specified in both info.json and config.h, the config.h values win.
avr-gcc (GCC) 5.4.0
```

Comment (CyberGear) [reply keyboards/cybergear/macro25/config.h]: @fauxpark is this good enough? :)

File: keyboards/cybergear/macro25/config.h [hunk 12]
```diff
@@ -0,0 +1,26 @@
+	/* Copyright 2021 CyberGear Marius Kavoliunas
```
Comment (fauxpark) [inline keyboards/cybergear/macro25/config.h:1]:
```suggestion
/* Copyright 2021 CyberGear Marius Kavoliunas
```

File: keyboards/cybergear/macro25/config.h [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define MANUFACTURER    CyberGear
+#define PRODUCT         Macro25 (mk-macro-01)
+
+#define MATRIX_ROWS 2
+#define MATRIX_COLS 5
```
Comment (fauxpark) [inline keyboards/cybergear/macro25/config.h:26]: Does it still compile properly without these? I'm unsure whether they can be computed from the number of pins specified in the info.json, or whether they must be added as `matrix_size`.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cybergear/macro25/config.h (hunk 13 above)
Comment (CyberGear) [reply keyboards/cybergear/macro25/config.h]: @fauxparYou were right, those two are not necessary.

**Review by fauxpark — APPROVED** [review]

File: keyboards/cybergear/macro25/info.json [hunk 14]
```diff
@@ -0,0 +1,25 @@
+{
+    "keyboard_name": "Macro225 (mk-macro-01)",
+    "url": "https://github.com/CyberGear/mk-macro-01",
+    "maintainer": "CyberGear",
+    "usb": {
+        "vid": "0x1209",
+        "pid": "0x69A1",
+        "device_ver": "0x0001"
+    },
+    "width": 10,
```
Comment (noroadsleft) [inline keyboards/cybergear/macro25/info.json:10]:
```suggestion
    "width": 5,
```

File: keyboards/cybergear/macro25/info.json [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+    "height": 2,
+    "diode_direction": "COL2ROW",
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/cybergear/macro25/info.json:14]: I have a preference for these macro names being specific.

```suggestion
        "LAYOUT_ortho_2x5": {
```

File: keyboards/cybergear/macro25/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+
+#define MAC(kc) (QK_RCTL | QK_RGUI | QK_RALT | (kc))
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/cybergear/macro25/keymaps/default/keymap.c:23]:
```suggestion
    [0] = LAYOUT_ortho_2x5(
```

File: keyboards/cybergear/macro25/keymaps/via/keymap.c [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+
+#define MAC(kc) (QK_RCTL | QK_RGUI | QK_RALT | (kc))
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/cybergear/macro25/keymaps/via/keymap.c:23]:
```suggestion
    [0] = LAYOUT_ortho_2x5(
```

File: keyboards/cybergear/macro25/keymaps/via/keymap.c [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+        MAC(KC_KP_0), MAC(KC_KP_1), MAC(KC_KP_2), MAC(KC_KP_3), MAC(KC_KP_4),
+        MAC(KC_KP_5), MAC(KC_KP_6), MAC(KC_KP_7), MAC(KC_KP_8), MAC(KC_KP_9)
+    ),
+    [1] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/cybergear/macro25/keymaps/via/keymap.c:27]:
```suggestion
    [1] = LAYOUT_ortho_2x5(
```

File: keyboards/cybergear/macro25/keymaps/via/keymap.c [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [2] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/cybergear/macro25/keymaps/via/keymap.c:31]:
```suggestion
    [2] = LAYOUT_ortho_2x5(
```

File: keyboards/cybergear/macro25/keymaps/via/keymap.c [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [3] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/cybergear/macro25/keymaps/via/keymap.c:35]:
```suggestion
    [3] = LAYOUT_ortho_2x5(
```

File: keyboards/cybergear/macro25/macro25.h [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/cybergear/macro25/macro25.h:21]:
```suggestion
#define LAYOUT_ortho_2x5( \
```

File: keyboards/cybergear/macro25/readme.md [hunk 22]
```diff
@@ -0,0 +1,15 @@
+# CyberGear Macro25
+
+![](https://i.imgur.com/Jk2xGMvh.jpg)
+
+Macro keyboard meade using arduino pro micro, with VIA support
```
Comment (noroadsleft) [inline keyboards/cybergear/macro25/readme.md:5]:
```suggestion
Macro keyboard made using Arduino Pro Micro, with VIA support.
```

File: keyboards/cybergear/macro25/readme.md [hunk 23]
```diff
@@ -0,0 +1,15 @@
...
+
+* Keyboard Maintainer: [CyberGeear](https://github.com/CyberGear)
+* Hardware Supported: arduino pro micro
+* Hardware Availability: [repo](https://github.com/CyberGear/mk-macro-01)
```
Comment (noroadsleft) [inline keyboards/cybergear/macro25/readme.md:9]:
```suggestion
* Keyboard Maintainer: [CyberGear](https://github.com/CyberGear)
* Hardware Supported: Arduino Pro Micro
* Hardware Availability: [GitHub](https://github.com/CyberGear/mk-macro-01)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/12518/files).

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**